community/kernel/src/main/java/org/neo4j/kernel/impl/store/counts/CountsStorageServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+    public Updater apply( long txId )
+    {
+        return updaterFor( txId );
+    }
+
+    @Override
+    public Register.DoubleLongRegister nodeCount( int labelId, Register.DoubleLongRegister target )
+    {
+        long value = countsStore.get( CountsKeyFactory.nodeKey( labelId ) )[0];
+        target.write( 0, value );
+        return target;
+    }
+
+    @Override
+    public Register.DoubleLongRegister relationshipCount( int startLabelId, int typeId, int endLabelId,
+            Register.DoubleLongRegister target )
+    {
+        long value = countsStore.get( CountsKeyFactory.relationshipKey( startLabelId, typeId, endLabelId ) )[0];
+        target.write( 0, value );
+        return target;
+    }
+
+    @Override
+    public Register.DoubleLongRegister indexUpdatesAndSize( int labelId, int propertyKeyId,
+            Register.DoubleLongRegister target )
+    {
+        long[] values = countsStore.get( CountsKeyFactory.indexStatisticsKey( labelId, propertyKeyId ) );
+        target.write( values[0], values[1] );
+        return target;
+    }
+
+    @Override
+    public Register.DoubleLongRegister indexSample( int labelId, int propertyKeyId, Register.DoubleLongRegister target )
+    {
+        long[] values = countsStore.get( CountsKeyFactory.indexSampleKey( labelId, propertyKeyId ) );
+        target.write( values[0], values[1] );
+        return target;
+    }
+
+    @Override
+    public void accept( CountsVisitor visitor )
+    {
+        countsStore.forEach( ( countsKey, values ) -> {
+            switch ( countsKey.recordType() )
+            {
+            case ENTITY_NODE:
+            {
+                visitor.visitNodeCount( ((NodeKey) countsKey).getLabelId(), values[0] );
+                break;
+            }
+            case ENTITY_RELATIONSHIP:
+            {
+                RelationshipKey k = (RelationshipKey) countsKey;
+                visitor.visitRelationshipCount( k.getStartLabelId(), k.getTypeId(), k.getEndLabelId(), values[0] );
+                break;
+            }
+            case INDEX_STATISTICS:
+            {
+                IndexStatisticsKey key = (IndexStatisticsKey) countsKey;
+                visitor.visitIndexStatistics( key.labelId(), key.propertyKeyId(), values[0], values[1] );
+                break;
+            }
+            case INDEX_SAMPLE:
+            {
+                IndexSampleKey key = (IndexSampleKey) countsKey;
+                visitor.visitIndexSample( key.labelId(), key.propertyKeyId(), values[0], values[1] );
+                break;
+            }
+            default:
+                throw new IllegalStateException( "unexpected counts key " + countsKey );
+            }
+        } );
+    }
+
+    @Override
+    public void initialize( CountsSnapshot snapshot, DatabaseHealth databaseHealth )
+    {
+        countsStore = new InMemoryCountsStore( snapshot, databaseHealth );
+        indexStatsUpdater = new IndexStatsUpdaterFactory().indexStatsUpdater( countsStore );
```

Comment:
Can `IndexStatsUpdaterFactory` be removed? Feels like we can create `indexStatsUpdater` directly.